modules/jenkins.test.js

```diff
@@ -0,0 +1,152 @@
+            const url = "aaa";
+            const username = "aaa";
+            const token = "";
+            const echoSpy = jest.spyOn(echo, "errors").mockResolvedValueOnce();
+
+            // Act
+            const result = jenkins.configureCredentials(url, username, token);
+
+            // Assert
+            expect(result).toBeFalse();
+            expect(echoSpy).toHaveBeenCalledTimes(1);
+        });
+
+        test("when url, username and token provided then returns true ", () => {
+            // Arrange
+            jenkins.writeToConfig = jest.fn(() => true);
+            const url = "aaa";
+            const username = "aaa";
+            const token = "aaa";
+
+            // Act
+            const result = jenkins.configureCredentials(url, username, token);
+
+            // Assert
+            expect(result).toBeTrue();
+        });
+    });
+
+    // #endregion configureCredentials
+
+    // -----------------------------------------------------------------------------------------
+    // #region getConfig
+    // -----------------------------------------------------------------------------------------
+
+    describe("getConfig", () => {
+        test("when config does not exist, then returns undefined", () => {
+            // Arrange
+            jest.spyOn(fs, "readFileSync").mockImplementation(
+                jest.fn(() => undefined)
+            );
+
+            // Act
+            const result = jenkins.getConfig();
+
+            // Assert
+            expect(result).toBeUndefined();
+        });
+
+        test("should return a JSON config object", () => {
+            // Arrange
+            jest.spyOn(fs, "readFileSync").mockImplementation(
+                jest.fn(() => JSON.stringify({ value: true }))
+            );
+
+            // Act
+            const result = jenkins.getConfig();
+
+            // Assert
+            expect(result).toBeObject();
+        });
+    });
+    // #endregion getConfig
+
+    // -----------------------------------------------------------------------------------------
+    // #region getConfigPath
+    // -----------------------------------------------------------------------------------------
+
+    describe("getConfigPath", () => {
+        test("should return a string ending in .jenkinsconfig", () => {
+            // Act
+            const result = jenkins.getConfigPath();
+
+            // Assert
+            expect(result).toContain(".jenkinsconfig");
+        });
+    });
+    // #endregion getConfigPath
+
+    // -----------------------------------------------------------------------------------------
+    // #region writeToConfig
+    // -----------------------------------------------------------------------------------------
+
+    describe("writeToConfig", () => {
+        test("when writeFileSyncSucceeds then returns true", () => {
+            // Arrange
+            jest.spyOn(fs, "writeFileSync").mockImplementation(
+                jest.fn(() => undefined)
```

Comment:
Minor - but I don't think you need the wrapping `jest.fn` here - passing in the function for the implementation should be enough, (ie `mockImplementation(() => undefined))`, or even just `mockImplementation()` should default to a function that returns `undefined` if I remember correctly)`